linux_os/guide/system/permissions/files/dir_perms_world_writable_root_owned/ansible/shared.yml

```diff
@@ -27,34 +26,57 @@
       - lustre
       - davfs
       - fuse.sshfs
-
-- name: "Create empty list of excluded paths"
-  set_fact:
     excluded_paths:
-      - /proc
-
-- name: "Detect nonlocal file systems and add them to excluded paths"
-  set_fact:
-    excluded_paths: "{{ excluded_paths | union([item.mount]) }}"
-  loop: "{{ ansible_mounts }}"
-  when: item.fstype in excluded_fstypes
-
-- name: "Find all directories excluding non-local partitions"
-  find:
-    paths: "/"
-    excludes: excluded_paths
+      - dev
+      - proc
+      - run
+      - sys
+    search_paths: []
+    world_writable_dirs: []
+
+- name: "{{{ rule_title }}} - Find Relevant Root Directories Ignoring Pre-Defined Excluded Paths"
+  ansible.builtin.find:
+    paths: /
     file_type: directory
-    hidden: yes
-    recurse: yes
-  register: found_dirs
+    excludes: "{{ excluded_paths }}"
+    hidden: true
+    recurse: false
+  register: result_relevant_root_dirs
+
+- name: "{{{ rule_title }}} - Include Relevant Root Directories in a List of Paths to be Searched"
+  ansible.builtin.set_fact:
+    search_paths: "{{ search_paths | union([item.path]) }}"
+  loop: "{{ result_relevant_root_dirs.files }}"
+
+- name: "{{{ rule_title }}} - Increment Search Paths List with Local Partitions Mount Points"
+  ansible.builtin.set_fact:
+    search_paths: "{{ search_paths | union([item.mount]) }}"
+  loop: '{{ ansible_mounts }}'
+  when:
+    - item.fstype not in excluded_fstypes
+    - item.mount != '/'
+
+- name: "{{{ rule_title }}} - Increment Search Paths List with Local NFS File System Targets"
+  ansible.builtin.set_fact:
+    search_paths: "{{ search_paths | union([item.device.split(':')[1]]) }}"
+  loop: '{{ ansible_mounts }}'
+  when:
+    - item.device is search("localhost:")
+
+- name: "{{{ rule_title }}} - Find All Uncompliant Directories in Local File Systems"
+  ansible.builtin.command:
+    cmd: find {{ item }} -xdev -type d -perm -0002 -uid +0
```

Comment:
We don't have something similar to `-xdev` (or -mount) in the `find` module and this was the simplest option for this specific case.